backend/src/main/java/jonaszeihe/ninjin/controller/CourseController.java

```diff
@@ -24,6 +26,12 @@ public Course addCourse(@RequestBody AddCourseDto dto) {
         return this.courseService.addCourse(dto.getName(), dto.getDuration());
     }
 
+    @GetMapping("{courseName}")
```

Comment:
i think your courseName is not unique. You removed the id annotation. Would be bettet to get the course by id and not by name